### PR TITLE
Add metamx emitter, http clients, and metrics packages to druid java-util

### PR DESCRIPTION
**Pull request by jon-wei**
This PR adds the emitter, http.client, and metrics packages from metamx's java-util to druid's own java-util module. 

This is tagged with Release Notes because this changes configuration properties to use `io.druid.java.util.*` names instead of `com.metamx.*`.

### CODE REVIEW

**Comment by jon-wei** [comment]
This PR will cause the sigar native libraries that were previously included in the com.metamx java-util jar to be included in the druid java-util jar.

If anyone has objections to this, please comment on this PR.

File: java-util/src/test/java/io/druid/java/util/http/client/io/AppendableByteArrayInputStreamTest.java [hunk 1]
```diff
@@ -72,7 +73,7 @@ public void testIndividualRead(byte[][] writtenBytes, byte[] expectedBytes) thro
     for (int i = 0; i < expectedBytes.length; i++) {
       final int readByte = in.read();
       if (expectedBytes[i] != (byte) readByte) {
-        Assert.assertEquals(String.format("%s[%d", Arrays.toString(expectedBytes), i), expectedBytes[i], readByte);
+        Assert.assertEquals(StringUtils.format("%s[%d", Arrays.toString(expectedBytes), i), expectedBytes[i], readByte);
```
Comment (leventov) [inline java-util/src/test/java/io/druid/java/util/http/client/io/AppendableByteArrayInputStreamTest.java]: Missing `]`

**Comment by leventov** [comment]
Also fix inspection problem

**Comment by gianm** [comment]
LGTM once the CI is passing.

**Comment by gianm** [comment]
Changed milestone since we did a backport.

**Pull request merged**